csrc/python_frontend/fusion_record.h

```diff
@@ -366,6 +366,57 @@ struct OpRecord : RecordFunctor {
   std::function<OutType(ArgTypes...)> fusion_op_;
 };
 
+struct SliceOpRecord : RecordFunctor {
+  SliceOpRecord(std::vector<State> _args, std::vector<State> _outputs)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.slice",
+            serde::RecordType::SliceOp) {
+    arg_names_[1] = "start_indices";
+    arg_names_[2] = "end_indices";
+    arg_names_[3] = "strides";
+  }
+  ~SliceOpRecord() override = default;
+  RecordFunctor* clone() final {
+    return new SliceOpRecord(*this);
+  }
+
+  void operator()(FusionState& fd) final {
+    TensorView* arg = fd.getFusionState(args_.at(0).index)->as<TensorView>();
+    const std::vector<Val*>& start = fd.getFusionStateVector(args_.at(1).index);
+    const std::vector<Val*>& end = fd.getFusionStateVector(args_.at(2).index);
+    const std::vector<Val*>& stride =
+        fd.getFusionStateVector(args_.at(3).index);
+    std::vector<Slice> vec_slice;
+    for (const auto idx : c10::irange(arg->nDims())) {
+      // NOTE: there's an extra move, we can use emplace_back if we go write
+      // some constructors for Slice.
```

Comment:
You could also do the following
```c++
vec_slice.emplace_back();
vec_slice.back().start = start_idx;
vec_slice.back().end = end_idx;
vec_slice.back().stride = stride_idx;
```
But I agree adding a constructor for `Slice` and using `vec_slice.emplace_back(start_idx, end_idx, stride_idx)` is simplest and clearest...

Reply:
I'm leaning towards leaving this as-is since it's just a move. I can have a follow up clean up if you think this bothers too much :laughing:

Reply:
> leaving this as-is

All good with me